tests/ArmTemplates.Tests/Extractor/Scenarios/ApiManagementServiceExtractorTests.cs

```diff
@@ -0,0 +1,107 @@
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Tests.Moqs.ApiClients;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Extensions;
+using Xunit;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Tests.Extractor.Scenarios
+{
+    [Trait("Category", "ApiManagementService Extraction")]
+    public class ApiManagementServiceExtractorTests : ExtractorMockerWithOutputTestsBase
+    {
+        public ApiManagementServiceExtractorTests() : base("api-management-service")
+        {
+        }
+
+        [Fact]
+        public async Task GenerateApiManagementService_ProperlyLaysTheInformation()
+        {
+            // arrange
+            var currentTestDirectory = Path.Combine(this.OutputDirectory, nameof(GenerateApiManagementService_ProperlyLaysTheInformation));
+
+            var extractorConfig = this.GetDefaultExtractorConsoleAppConfiguration();
+
+            var extractorParameters = new ExtractorParameters(extractorConfig);
+
+            var mockedApiManagementServiceClient = MockApiManagementServiceClient.GetMockedApiManagementServiceClientWithDefaultValues();
+
+            var mockedApiManagementServiceExtractor = new ApiManagementServiceExtractor(this.GetTestLogger<ApiManagementServiceExtractor>(), new TemplateBuilder(), mockedApiManagementServiceClient);
+            
+            var extractorExecutor = ExtractorExecutor.BuildExtractorExecutor(
+                this.GetTestLogger<ExtractorExecutor>(),
+                apiManagementServiceExtractor: mockedApiManagementServiceExtractor);
+            extractorExecutor.SetExtractorParameters(extractorParameters);
+
+            // act
+            var apiManagementServiceTemplate = await extractorExecutor.GenerateApiManagementServiceTemplate(currentTestDirectory);
+
+            // assert
+            File.Exists(Path.Combine(currentTestDirectory, extractorParameters.FileNames.ApiManagementService)).Should().BeTrue();
+
+            apiManagementServiceTemplate.Parameters.Should().NotBeNull();
+            apiManagementServiceTemplate.Parameters.Should().ContainKey(ParameterNames.ApimServiceName);
+            apiManagementServiceTemplate.Resources.Count().Should().Be(1);
+            apiManagementServiceTemplate.TypedResources.ApiManagementServices.Count().Should().Be(1);
+        }
+
+        [Fact]
+        public async Task GenerateApiManagementService_ProperlyParsesJsonResponse()
+        {
+            var currentTestDirectory = Path.Combine(this.OutputDirectory, nameof(GenerateApiManagementService_ProperlyParsesJsonResponse));
+            var extractorConfig = this.GetDefaultExtractorConsoleAppConfiguration();
+            var extractorParameters = new ExtractorParameters(extractorConfig);
+
+            FileReader fileReader = new FileReader();
+            string fileLocation = Path.Combine("Resources", "ApiClientJsonResponses", "api_management_success_response.json");
```

Comment:
Couple of static methods introduced in the MockClientUtils class, deserializer of the content, reader and main httpClientFactory mock.

Reply:
Sorry, my bad in terms of poor explanation: I wanted to ask to not have hardcoded path to test-json's.
This is a great idea to determine proper folder structure like `Resources\ApiClientJsonResponses\<json-name>.json`, but it will be pretty cool to have them stored somewhere in base class, so that we will not be writing that every time in every class.

In our internal solution we have such a code for testing some functionality with Json's, and all we do in the test itself is specify the name of file, not the full path like `Resources` and `ApiClientJsonResponses`

Reply:
Got it, Resources path now is const string with no possibility to change, so every Test should provide location of file inside the resources.